genie-web/src/main/java/com/netflix/genie/web/tasks/leader/ClusterCheckerTask.java

```diff
@@ -109,70 +121,99 @@ public ClusterCheckerTask(
+        final boolean isNodeHealthy = isNodeHealthy(host);
+        //
+        // If node is healthy, remove the entry from the errorCounts.
+        // If node is not healthy, update the entry in errorCounts
+        //
+        if (isNodeHealthy) {
+            if (errorCounts.containsKey(host)) {
+                errorCounts.remove(host);
+            }
+        } else {
+            if (this.errorCounts.containsKey(host)) {
+                this.errorCounts.put(host, this.errorCounts.get(host) + 1);
+            } else {
+                this.errorCounts.put(host, 1);
+            }
+        }
+    }
 
-        // Did we pass bad threshold on any hosts? Error jobs if so
-        toRemove.clear();
-        this.errorCounts.keySet()
-            .stream()
-            .filter(host -> this.errorCounts.get(host) == this.properties.getLostThreshold())
-            .forEach(
-                host -> {
-                    toRemove.add(host);
-                    final Set<Job> jobs = this.jobSearchService.getAllActiveJobsOnHost(host);
-                    jobs.forEach(
-                        job -> {
-                            try {
-                                this.jobPersistenceService.setJobCompletionInformation(
-                                    job.getId(),
-                                    JobExecution.LOST_EXIT_CODE,
-                                    JobStatus.FAILED,
-                                    "Genie leader can't reach node running job. Assuming node and job are lost."
-                                );
-                                this.lostJobsCounter.increment();
-                            } catch (final GenieException ge) {
-                                log.error("Unable to update job {} to failed due to exception", job.getId(), ge);
-                                this.unableToUpdateJobCounter.increment();
-                            }
-                        }
-                    );
+    private boolean isNodeHealthy(final String host) {
+        //
+        // A node is valid and healthy if all health endpoints excluding the ones mentioned in ignoreHealthEndPoints
+        // are UP.
+        //
+        boolean result = true;
+        try {
+            restTemplate.getForObject(this.scheme + host + this.healthEndpoint, String.class);
+        } catch (final HttpStatusCodeException e) {
+            log.error("Failed validating host {}", host, e);
+            try {
+                final Map<String, Object> responseMap = mapper.readValue(e.getResponseBodyAsByteArray(),
+                    TypeFactory.defaultInstance().constructMapType(Map.class, String.class, Object.class));
+                for (Map.Entry<String, Object> responseEntry : responseMap.entrySet()) {
+                    if (responseEntry.getValue() instanceof Map
+                        && !ignoreHealthEndPoints.contains(responseEntry.getKey())
+                        && Status.OUT_OF_SERVICE.getCode()
```

Comment:
I think you need to check if it's just not UP... basically any state other than that is bad. Looking at the internal Healthchecks we'll probably want to add discovery to the ones to ignore due to it being in UNKNOWN state but we don't care.